src/core/primitives.js

```diff
@@ -383,6 +383,14 @@ class RefSetCache {
     this._map.clear();
   }
 
+  isEmpty() {
+    return this._map.size === 0;
+  }
+
+  *values() {
+    yield* this._map.values();
+  }
+
```

Comment:
Sorry, but why do we need two new methods (that are only used *once* each) when the same functionality *already exists* in this class!?

For example, it looks trivial to change the call-sites as follows:
 - `else if (changes.isEmpty()) { ... }` -> `else if (!changes.size) { ... }`
 - `changes.values().some(...)` -> `[...changes].some(...)`

Reply:
- `changes.isEmpty` is used in some unit tests too
- about `values`, I found that it was a pity to create an intermediary array (allocation, pushing, growing, ...) when we almost just want to iterate on the values.

Reply:
> * `changes.isEmpty` is used in some unit tests too

Sure, but all of those can be *trivially* replaced by the existing `size` getter (to avoid needless code-size increase).

> * about `values`, I found that it was a pity to create an intermediary array

OK, let's keep that one then.